src/raftstore/coprocessor/dispatcher.rs

```diff
@@ -47,10 +53,18 @@ pub struct CoprocessorHost {
 }
 
 impl CoprocessorHost {
-    // TODO load from configuration.
-
-    pub fn new() -> CoprocessorHost {
-        CoprocessorHost::default()
+    pub fn new<C: Sender<Msg> + Send + Sync + 'static>(
+        cfg: Config,
+        ch: RetryableSendCh<Msg, C>,
+    ) -> CoprocessorHost {
+        let mut registry = Registry::default();
+        let split_size_check_observer =
+            SizeCheckObserver::new(cfg.region_max_size.0, cfg.region_split_size.0, ch);
+        registry.register_observer(100, Box::new(split_size_check_observer));
+        if cfg.split_region_on_table {
+            registry.register_observer(99, Box::new(TableCheckObserver::default()));
+        }
+        CoprocessorHost { registry: registry }
```

Comment:
can we guarantee the table observer can be run before size observer?

Reply:
Yes, the priority of table observer is higher[0] than size observer.

[0] higher means less.